src/Microsoft.AspNetCore.WebHooks.Receivers/Filters/WebHookVerifyBodyTypeFilter.cs

```diff
@@ -125,6 +132,74 @@ public void OnResourceExecuted(ResourceExecutedContext context)
             // No-op
         }
 
+        /// <summary>
+        /// Determines whether the specified request contains JSON as indicated by a content type of
+        /// <c>application/json</c>, <c>text/json</c> or <c>application/xyz+json</c>. The term <c>xyz</c> can for
+        /// example be <c>hal</c> or some other JSON-derived media type.
+        /// </summary>
+        /// <param name="request">The <see cref="HttpRequest"/> to check.</param>
+        /// <returns>
+        /// <see langword="true"/> if the specified request contains JSON content; otherwise, <see langword="false"/>.
+        /// </returns>
+        protected static bool IsJson(HttpRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var contentType = request.GetTypedHeaders().ContentType;
+            if (contentType == null)
+            {
+                return false;
+            }
+
+            if (contentType.IsSubsetOf(ApplicationJsonMediaType) || contentType.IsSubsetOf(TextJsonMediaType))
+            {
+                return true;
+            }
+
+            // MVC's JsonInputFormatter does not support text/*+json by default. RFC 3023 and 6839 allow */*+json but
+            // https://www.iana.org/assignments/media-types/media-types.xhtml shows all +json registrations except
+            // model/gltf+json match application/*+json.
```

Comment:
> FYI, this is because the [RFC](https://tools.ietf.org/html/rfc6839#section-3.1) doesn't mention text/json and text/xml. The suffix content type feature specifically applies to application/json and application/xml (among others) specifically

I've updated this comment.

More generally, I'm not sure limiting ourselves to `application/*+json` and `application/*+xml` is correct according to [RFC 6839](https://tools.ietf.org/html/rfc6839). That and [RFC 3023](https://tools.ietf.org/html/rfc3023) talk more in terms of `*/*+json` and `*/*+xml` though I agree media types registered with suffixes (so far) almost all match `application/*+json` or `application/*+xml`. (The current exceptions match `message/*+xml`, `model/*+xml` and `model/gltf+json`.)

Thoughts about filing an aspnet/Http or aspnet/Mvc bug to further generalize of suffix handling?

Reply:
If you want to create a paper trail, feel free to create one. I think we'd decide to close it.

We register default content types for the common use cases, not all possible cases. Application developers can do whatever they want 👍